### PR TITLE
build: Fix Windows installer build

### PR DESCRIPTION
**Pull request by achow101**
Apparently NSIS requires a 4 digit version number, and #20223 dropped the 4th digit. So this adds a 4th 0 digit so that building the Windows installer doesn't fail. Also fixes a typo in that version string that was also present in a plist file.

### CODE REVIEW

File: share/qt/Info.plist.in [hunk 1]
```diff
@@ -17,7 +17,7 @@
   <string>APPL</string>
 
   <key>NSHumanReadableCopyright</key>
-  <string>@CLIENT_VERSION_MAJOR@.@CLIENT_VERSION_MINOR@.@.@CLIENT_VERSION_BUILD@, Copyright © 2009-@COPYRIGHT_YEAR@ @COPYRIGHT_HOLDERS_FINAL@</string>
+  <string>@CLIENT_VERSION_MAJOR@.@CLIENT_VERSION_MINOR@.@CLIENT_VERSION_BUILD@, Copyright © 2009-@COPYRIGHT_YEAR@ @COPYRIGHT_HOLDERS_FINAL@</string>
```
Comment (hebasto) [inline share/qt/Info.plist.in:20]: It seems the PR description says nothing about changes in this line, no?

**Comment by achow101** [comment]
```
d03d8552ac2aa67cf6a5fe414a659d63e9f4441a20be54545b6f78bafeb47c07  bitcoin-68c2ef13e953-win-unsigned.tar.gz
6fe15d137e9310fc25ccc86d10c6de3e35f34196153df020cb9843731b56f75d  bitcoin-68c2ef13e953-win64-debug.zip
c751fc2c0ce46324d988faaf5f1c98da0f66e96665ed6f7912a6ee1c6f641aba  bitcoin-68c2ef13e953-win64-setup-unsigned.exe
d55f40434f180ed462ff803787cc781c4700b6187b022449d99839ff16b86b06  bitcoin-68c2ef13e953-win64.zip
a9cdd730e924267ffcab275e02482221fb336a792747d518d207a8298983c0a1  src/bitcoin-68c2ef13e953.tar.gz
db03c292ae3f69b6b099f8c3306016336f21f7b226fd0757a8cf5b8ccb45bd71  bitcoin-core-win-22-res.yml
```

**Comment by fanquake** [comment]
ACK 68c2ef13e953f142f818a84be9e2c09ef8636ccc

> Apparently NSIS requires a 4 digit version number,

There's no mention in the [docs](https://nsis.sourceforge.io/Reference/VIProductVersion), but for reference here is the build failure:
```bash
STRIPPROG="/home/ubuntu/wrapped/x86_64-w64-mingw32-strip" /bin/bash /home/ubuntu/build/bitcoin/distsrc-x86_64-w64-mingw32/build-aux/install-sh -c -s ./src/bitcoin-wallet.exe ./release
Error: invalid VIProductVersion format, should be X.X.X.X
Error - aborting creation process
error: could not build /home/ubuntu/out/bitcoin-e9a1c9fbdea9-win64-setup-unsigned.exe
built /home/ubuntu/out/bitcoin-e9a1c9fbdea9-win64-setup-unsigned.exe
+ make install DESTDIR=/home/ubuntu/build/bitcoin/distsrc-x86_64-w64-mingw32/installed/bitcoin-e9a1c9fbdea9
```
Seems a bit arbitrary that they'd enforce 4 digit version numbers.

Also checked that the broken macOS version is fixed. i.e master is currently:
broken_version


Looks deterministic to me:
```bash
Generating report
d03d8552ac2aa67cf6a5fe414a659d63e9f4441a20be54545b6f78bafeb47c07  bitcoin-68c2ef13e953-win-unsigned.tar.gz
6fe15d137e9310fc25ccc86d10c6de3e35f34196153df020cb9843731b56f75d  bitcoin-68c2ef13e953-win64-debug.zip
c751fc2c0ce46324d988faaf5f1c98da0f66e96665ed6f7912a6ee1c6f641aba  bitcoin-68c2ef13e953-win64-setup-unsigned.exe
d55f40434f180ed462ff803787cc781c4700b6187b022449d99839ff16b86b06  bitcoin-68c2ef13e953-win64.zip
a9cdd730e924267ffcab275e02482221fb336a792747d518d207a8298983c0a1  src/bitcoin-68c2ef13e953.tar.gz
d55e10bc81e5fb89a1252633475c770e206cc38fb117bd8154d5aec57502c239  bitcoin-core-win-22-res.yml
Done.
```

**Review by hebasto — CHANGES_REQUESTED** [review]
Approach ACK 68c2ef13e953f142f818a84be9e2c09ef8636ccc, tested on Linux Mint 20 (x86_64, nsis 3.05-2).

While the `share/setup.nsi.in` is touched, maybe add the `Unicode True` line to fix the [warning](https://sourceforge.net/p/nsis/mailman/message/36536660/):
```
...
/bin/mkdir -p ./release
STRIPPROG="/usr/bin/x86_64-w64-mingw32-strip" /bin/bash /home/hebasto/GitHub/bitcoin/build-aux/install-sh -c -s ./src/bitcoind.exe ./release
STRIPPROG="/usr/bin/x86_64-w64-mingw32-strip" /bin/bash /home/hebasto/GitHub/bitcoin/build-aux/install-sh -c -s ./src/qt/bitcoin-qt.exe ./release
STRIPPROG="/usr/bin/x86_64-w64-mingw32-strip" /bin/bash /home/hebasto/GitHub/bitcoin/build-aux/install-sh -c -s ./src/bitcoin-cli.exe ./release
STRIPPROG="/usr/bin/x86_64-w64-mingw32-strip" /bin/bash /home/hebasto/GitHub/bitcoin/build-aux/install-sh -c -s ./src/bitcoin-tx.exe ./release
STRIPPROG="/usr/bin/x86_64-w64-mingw32-strip" /bin/bash /home/hebasto/GitHub/bitcoin/build-aux/install-sh -c -s ./src/bitcoin-wallet.exe ./release
warning 7998: ANSI targets are deprecated

1 warning:
  7998: ANSI targets are deprecated
built bitcoin-21.99.0-win64-setup.exe
```

**Comment by MarcoFalke** [comment]
@hebasto This should be addressed in a separate issue. See #13817

**Comment by laanwj** [comment]
> There's no mention in the docs, but for reference here is the build failure:

IIRC it's a windows thing in how version numbers are represented in the record in question, not so much a NSIS thing.

Code review ACK 68c2ef13e953f142f818a84be9e2c09ef8636ccc

**Pull request merged**